### PR TITLE
Removed profiles that were redundant with Cumulus code

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Please write a sentence or two summarizing the proposed change.

Addresses [ORCA-239: Remove provider block from all terraform modules and consolidate to main file.](https://bugs.earthdata.nasa.gov/browse/ORCA-239)

## Changes

* Removed redundant provider blocks.
* Made aws_profile optional.

## How Has This Been Tested?

Deployed to AWS.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [ ] CHANGELOG.md
    - [ ] Testing documentation
    - [ ] Development documentation
    - [ ] User documentation
    - [ ] (Additional here)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
    - [ ] Unit tests
    - [ ] Integration tests
    - [ ] Manual tests (outlined above)
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

File: modules/orca/variables.tf [hunk 1]
```diff
@@ -56,6 +50,13 @@ variable "workflow_config" {
 
 
 ## OPTIONAL - Default variable value is set in ../variables.tf to keep default values centralized.
+variable "aws_profile" {
+  type        = string
+  description = "AWS profile used to deploy the terraform application."
+  default = null
```
Comment (bhazuka) [inline modules/orca/variables.tf:56]: We would want to set this default in the base variables.tf file in the base of the repo.

**Review by bhazuka — CHANGES_REQUESTED** [review]
You will also need to touch the main.tf and variables.tf files in the base of the repo. In addition, you will want to update the documentation in the website/docs/developer/deployment-guide/deployment-with-cumulus.md file.

File: modules/rds/main.tf [hunk 2]
```diff
@@ -9,19 +9,12 @@ terraform {
...
-}
-
-
 ## Local Variables
 locals {
   tags = merge(var.tags, { Deployment = var.prefix })
   # Ignore aws profile in case this was deployed with CI or in a machine without
   # aws profile defined
-  used_profile = contains([var.aws_profile], "default") ? "" : "--profile ${var.aws_profile}"
+  used_profile = contains(["${var.aws_profile == null ? "" : var.aws_profile}"], "default") ? "" : "--profile ${var.aws_profile}"
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply modules/rds/main.tf:17]: That would go against the initial design of this line. The only purpose is to remove "default" and replace with "". Must be okay.

File: modules/orca/variables.tf (hunk 1 above)
Comment (AndrewDorn) [reply modules/orca/variables.tf:56]: Didn't even know that was a thing... Don't know why that isn't in the folder with the rest of the TF. Fixed.

**Comment by AndrewDorn** [comment]
> Also remove the variables from website/docs/developer/deployment-guide/deployment-with-cumulus.md

I have updated the file, but I can't remove those variables; They're still in use.

File: website/docs/developer/deployment-guide/deployment-with-cumulus.md [hunk 3]
```diff
@@ -125,6 +124,8 @@ The variables must be set with the proper values in the `terraform.tfvavrs` file
 
 :::note Optional Cumulus Values
 
+aws_profile should be set if using an aws_profile other than `default` for Cumulus' deployment.
```
Comment (bhazuka) [inline website/docs/developer/deployment-guide/deployment-with-cumulus.md:127]: This isn't very clear. I guess I am not sure what you are trying to convey here.

**Review by bhazuka — COMMENTED** [review]
Overall looks good, just some final questions.

File: modules/orca/variables.tf (hunk 1 above)
Comment (bhazuka) [reply modules/orca/variables.tf]: Has to do with the way the build occurs and making sure all the pathing is correct to the various resources.

**Review by bhazuka — APPROVED** [review]

**Pull request merged**